app/services/katello/pulp3/smart_proxy_repository.rb

```diff
@@ -33,6 +33,38 @@ def delete_orphaned_repositories_for_mirror_proxies
           ::Katello::Pulp3::Repository.new(nil, smart_proxy).repositories_api.delete(repo._href)
         end
       end
+
+      def pulp3_enabled_repo_types(content_types = SETTINGS[:katello][:content_types].keys)
+        repo_types = content_types.collect do |content_type|
+          Katello::RepositoryTypeManager.find(content_type)
+        end
+
+        repo_types.select { |type| type.try(:pulp3_service_class) }
```

Comment:
I think you want to make this method 'smart proxy' aware.  So instead of just assuming that all types that katello supports on pulp3 with pulp3 are actually deployed to the pulp3 server, you can use:
```
smart_proxy.pulp3_repository_type_support?(repository_type)
```

Reply:
Thanks!